### PR TITLE
Add airmass plot limits and regions

### PR DESCRIPTION
**Pull request by ejeschke**
Added kwarg parameters `lo_limit`/`lo_region` and `hi_limit`/`hi_region` to the `plot_airmass()` function.  `lo_limit` and `hi_limit` allow you to easily override the default 1.0/3.0 limits for the y-axis.  `lo_region` defaults to `None`, but if set, it will shade a region between `lo_limit` and `lo_region`.  `hi_region` performs a similar role for `hi_limit`.

These are useful to indicate problematic airmass ranges that you might want to avoid when observing.

### CODE REVIEW

**Comment by ejeschke** [comment]
Hey @bmorris3, not sure why this wanted to make a merge commit instead of a ff when I tried to update from upstream, but if you want you can just git cherry-pick the top commit to avoid any issues.

**Comment by ejeschke** [comment]
Example:
```python
import numpy as np
upper_lim_deg = 80.0
lo_region = 1./np.cos(np.radians(90 - upper_lim_deg))
lower_lim_deg = 30.0
hi_region = 1./np.cos(np.radians(90 - lower_lim_deg))

from astroplan.plots import plot_airmass
import matplotlib.pyplot as plt
from astropy.time import Time
from astroplan import FixedTarget, Observer

time = Time.now()
target = FixedTarget.from_name('HD 189733')
apo = Observer.at_site('APO')
plot_airmass(target, apo, time, brightness_shading=True, altitude_yaxis=True,
             lo_limit=0.95, lo_region=lo_region, hi_limit=2.5, hi_region=hi_region)
plt.show()
```

astroplan_limits

**Comment by ejeschke** [comment]
Travis errors seem to be from sphinx build.  Not sure what the issue is.

**Comment by bmorris3** [comment]
@ejeschke The issue seems to come from [this line](https://travis-ci.org/astropy/astroplan/jobs/119134710#L806):
```
WARNING: py:obj reference target not found: hi_region
```
Is this something we should fix with `nitpick`, as @embray suggested on [stackoverflow](http://stackoverflow.com/a/30624034/1340208)? Maybe Sphinx-whisperer @bsipocz has advice?

File: astroplan/plots/time_dependent.py [hunk 1]
```diff
@@ -101,6 +102,20 @@ def plot_airmass(target, observer, time, ax=None, style_kwargs=None,
         Add alternative y-axis on the right side of the figure with target
         altitude. Default is `False`.
 
+    lo_limit : float
+        Lower limit of y-axis airmass range in the plot. Default is `1.0`.
+
+    lo_region : float
+        If set, defines an interval between `lo_limit` and `lo_region`
```
Comment (bsipocz) [inline astroplan/plots/time_dependent.py]: kwarg names be double backticks as sphinx tries to linkify the singe ones and with fail with these.

File: astroplan/plots/time_dependent.py [hunk 2]
```diff
@@ -101,6 +102,20 @@ def plot_airmass(target, observer, time, ax=None, style_kwargs=None,
         Add alternative y-axis on the right side of the figure with target
         altitude. Default is `False`.
 
+    lo_limit : float
+        Lower limit of y-axis airmass range in the plot. Default is `1.0`.
```
Comment (bsipocz) [inline astroplan/plots/time_dependent.py]: Either omit the ticks, or use double ones

File: astroplan/plots/time_dependent.py [hunk 3]
```diff
@@ -101,6 +102,20 @@ def plot_airmass(target, observer, time, ax=None, style_kwargs=None,
...
+
+    lo_region : float
+        If set, defines an interval between `lo_limit` and `lo_region`
+        that will be shaded. Default is `None`.
+
+    hi_limit : float
+        Upper limit of y-axis airmass range in the plot. Default is `3.0`.
```
Comment (bsipocz) [inline astroplan/plots/time_dependent.py]: Either omit the ticks, or use double ones

File: astroplan/plots/time_dependent.py [hunk 4]
```diff
@@ -101,6 +102,20 @@ def plot_airmass(target, observer, time, ax=None, style_kwargs=None,
...
+
+    hi_region : float
+        If set, defines an interval between `hi_limit` and `hi_region`
```
Comment (bsipocz) [inline astroplan/plots/time_dependent.py]: double ticks around kwargs

**Comment by bsipocz** [comment]
@ejeschke @bmorris3 - The sphinx build should pass once all the 6 cases of singe backtick use is changed to use double ones. (Singe ones means links for sphinx, but there is nothing to link those to, so it raises the warnings).

**Pull request closed without merging**